lib/Parse/ParseDecl.cpp

```diff
@@ -7944,8 +7948,24 @@ static ParserStatus parseAccessorIntroducer(Parser &P,
     Status.setIsParseError();
     return Status;
   }
-#define SUPPRESS_ARTIFICIAL_ACCESSORS 1
 #define ACCESSOR_KEYWORD(KEYWORD)
+// Existing source code may have an implicit getter whose first expression is a
+// call to a function named experimental_accessor passing a trailing closure.
+// To maintain compatibility, if the feature corresponding to
+// experimental_accessor is not enabled, do not parse the token as an
+// introducer, so that it can be parsed as such a function call.
+#define EXPERIMENTAL_COROUTINE_ACCESSOR(ID, KEYWORD, FEATURE)                  \
+  else if (P.Tok.getRawText() == #KEYWORD) {                                   \
+    if (!P.Context.LangOpts.hasFeature(Feature::FEATURE)) {                    \
+      if (featureUnavailable)                                                  \
+        *featureUnavailable = true;                                            \
+      if (isFirstAccessor) {                                                   \
```

Comment:
Shouldn’t this be the other way round? Ie. shouldn’t we allow these tokens on the first accessor but not on any following? Or am I missing something?

Reply:
No, we need to allow `read` and `modify` to be the names functions called with an implicit return from an implicit getter.  See `test/Parse/coroutine_accessors_ambiguity.swift`.

Reply:
Oh, do we backtrack if we return an error status from `parseAccessorIntroducer`?

Reply:
Yeah, in `parseGetSet`.

Reply:
Great. Thanks for confirming.